Code/Framework/AzToolsFramework/Tests/Prefab/PrefabTestDomUtils.cpp

```diff
@@ -166,13 +166,29 @@ namespace UnitTest
             if (expectedNestedInstanceDomInstances.has_value())
             {
                 ASSERT_TRUE(actualNestedInstanceDomInstances.has_value());
-                for (auto instanceIterator = expectedNestedInstanceDomInstances->get().MemberBegin();
-                     instanceIterator != expectedNestedInstanceDomInstances->get().MemberEnd(); ++instanceIterator)
+                if (expectedNestedInstanceDomInstances->get().IsArray())
```

Comment:
Nested instances are always stored as a map in prefabs(except for empty values I think; need to check). So, I don't think this would be needed. Is there any reason for adding this now?

Reply:
yeah, MemberBegin was producing an assert in rapidjson because the thing inside it was not a map
I added the case for array because the thing inside was an array, in all the cases I debugged, the array was empty

Reply:
I think the array is always empty because json serializer defaults an empty map to be an array. This is only showing up now because we changed the default behavior of prefab loading to always keep the default values(instead of empty objects). So, if nested instances is an array, all we need to check is this
```
ASSERT_TRUE(actualNestedInstanceDomInstances->get().IsArray());
const size_t expectedArraySize= expectedNestedInstanceDomInstances->get().GetArray().Size();
const size_t actualArraySize= actualNestedInstanceDomInstances->get().GetArray().Size();
ASSERT_TRUE(expectedArraySize == 0 && actualArraySize == 0);
```

Reply:
Ok, I can change that

Reply:
If you do don't use ASSERT_TRUE for the latter checks.
Use `EXPECT_EQ` to provide better error messaging.
`ASSERT_TRUE` can only provide the expression in the error message, while the *_EQ function can resolve the value of the variables in each expression
```
EXPECT_EQ(0, expectedArraySize);
EXPECT_EQ(0, actualArraySize);
```